packages/rest/test/acceptance/routing/routing.acceptance.ts

```diff
@@ -499,7 +525,7 @@ describe('Routing', () => {
   }
 
   function givenControllerInApp(app: Application, controller: ControllerClass) {
-    app.controller(controller);
+    app.controller(controller).inScope(BindingScope.CONTEXT);
```

Comment:
IMO, all controllers should be scoped to the request context. Can you think of a use case where we would want multiple controller instances in a single request context?

Originally, I wanted to propose to add `inScope` call to `app.controller()`. On the second thought, I am not sure if it's a good idea - what happens when somebody calls `app.get('controllers.ProductController')`, will future calls of `requestContext.get('controllers.ProductController')` return the value cached at app-level? Maybe we need a new scope called `CHILD_CONTEXT`?

Reply:
The `CONTEXT` scope should be the default for controllers. It is designed to work nicely with your use case:

1. `app.get('controllers.ProductController')` --> create and cache an instance at `app` level
2. `requestContext.get('controllers.ProductController')` --> Now it's the req context and a new instance will be created and cached at `requestContext` level
3. `requestContext.get('controllers.ProductController')` --> Hits the cache, get the same instance as 2
4. `anotherRequestContext.get('controllers.ProductController')` --> Now it's another req context and a new instance will be created and cached at `anotherRequestContext` level

Reply:
Cool. In that case, I am proposing to:

 - Call `inScope` from `app.controller()` to automatically register all controllers in CONTEXT scope.
 - Add a test to verify the behavior described in your comment. For example:
    1. Register a controller class that stores state in the instance.
    2. Call `app.get`, mutate the state stored in the controller instance.
    3. Make an HTTP request (this will get controller instance from the request context), verify that the response contains original (not mutated) state.